authlib/jose/rfc7516/jwe.py

```diff
@@ -197,6 +338,110 @@ def deserialize_compact(self, s, key, decode=None, sender_key=None):
             payload = decode(payload)
         return {'header': protected, 'payload': payload}
 
+    def deserialize_json(self, obj, key, decode=None, sender_key=None):
```

Comment:
In many cases we will need to parse the message first, get the recipient key IDs, and only then call descrypt/deserialize for the found recipient key values. So, we will need a separate method to parse the message.

Reply:
Added `parse_json` method.